src/lib/TopLevel.hs

```diff
@@ -187,20 +172,20 @@ initializeBackend backend = case backend of
   _ -> error "Not implemented"
 
 arrayVars :: HasVars a => a -> [Var]
-arrayVars x = map (\(v@(Name ArrayName _ _), L ty) -> v :> ty) $ envPairs (freeVars x)
+arrayVars x = [v:>ty | (v@(Name ArrayName _ _), ty) <- envPairs (freeVars x)]
```

Comment:
I prefer the old implementation because it made the existence of any non-array free variable a crash, whereas this one will happy skip such vars.

Reply:
Oh good point. In the merge I read them as identical and picked an arbitrary one.